contracts/0.8.25/vaults/Permissions.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+
+// See contracts/COMPILERS.md
+pragma solidity 0.8.25;
+
+import {AccessControlVoteable} from "contracts/0.8.25/utils/AccessControlVoteable.sol";
+import {OwnableUpgradeable} from "contracts/openzeppelin/5.2/upgradeable/access/OwnableUpgradeable.sol";
+
+import {IStakingVault} from "./interfaces/IStakingVault.sol";
+import {VaultHub} from "./VaultHub.sol";
+import {ILido as IStETH} from "../interfaces/ILido.sol";
+
+/**
+ * @title Permissions
+ * @author Lido
+ * @notice Provides granular permissions for StakingVault operations.
+ */
+abstract contract Permissions is AccessControlVoteable {
+    /**
+     * @notice Permission for funding the StakingVault.
+     */
+    bytes32 public constant FUND_ROLE = keccak256("StakingVault.Permissions.Fund");
+
+    /**
+     * @notice Permission for withdrawing funds from the StakingVault.
+     */
+    bytes32 public constant WITHDRAW_ROLE = keccak256("StakingVault.Permissions.Withdraw");
+
+    /**
+     * @notice Permission for minting stETH shares backed by the StakingVault.
+     */
+    bytes32 public constant MINT_ROLE = keccak256("StakingVault.Permissions.Mint");
+
+    /**
+     * @notice Permission for burning stETH shares backed by the StakingVault.
+     */
+    bytes32 public constant BURN_ROLE = keccak256("StakingVault.Permissions.Burn");
+
+    /**
+     * @notice Permission for rebalancing the StakingVault.
+     */
+    bytes32 public constant REBALANCE_ROLE = keccak256("StakingVault.Permissions.Rebalance");
+
+    /**
+     * @notice Permission for requesting validator exit from the StakingVault.
+     */
+    bytes32 public constant REQUEST_VALIDATOR_EXIT_ROLE = keccak256("StakingVault.Permissions.RequestValidatorExit");
+
+    /**
+     * @notice Permission for voluntary disconnecting the StakingVault.
+     */
+    bytes32 public constant VOLUNTARY_DISCONNECT_ROLE = keccak256("StakingVault.Permissions.VoluntaryDisconnect");
+
+    function _stakingVault() internal view virtual returns (IStakingVault);
```

Comment:
Can we store the address in this contracts without this virtual methods overriding?